src/main/scala/com/weather/scalacass/CassFormatDecoder.scala

```diff
@@ -121,13 +125,16 @@ object CassFormatDecoder extends CassFormatDecoderVersionSpecific {
       }
       process(f.listIterator)
     }
-    def extract(r: Row, name: String) = r getList (name, underlying.clazz)
-    def tupleExtract(tup: TupleValue, pos: Int) = tup getList (pos, underlying.clazz)
+    def extract(r: Row, name: String) = r getList (name, underlying.typeToken)
+    def tupleExtract(tup: TupleValue, pos: Int) = tup getList (pos, underlying.typeToken)
   }
 
+  private def setOf[T](eltType: TypeToken[T]): TypeToken[java.util.Set[T]] =
```

Comment:
without this (and `listOf` and `mapOf`), the type inferer gets things wrong, and there's no way to explicitly set the type. as far as i can tell, it is impossible without these functions